include/aos/common/resourcemonitor.hpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_RESOURCEMONITOR_HPP_
+#define AOS_RESOURCEMONITOR_HPP_
+
+#include "aos/common/tools/error.hpp"
+#include "aos/common/types.hpp"
+
+namespace aos {
+
+/**
+ * Resource usage provider interface.
+ */
+class ResourceUsageProviderItf {
+public:
+    /**
+     * Gathering system info
+     *
+     * @param systemInfo system info
+     * @return Error
+     */
+    virtual Error GatheringSystemInfo(SystemInfo* systemInfo) const = 0;
+
+    /**
+     * Gathering system data
+     *
+     * @param monitoringData monitoring data
+     * @return Error
+     */
+    virtual Error GatheringSystemData(NodeMonitoringData* monitoringData) const = 0;
+
+    /**
+     * Gathering instance data
+     *
+     * @param instance instance ident
+     * @param monitoringData monitoring data
+     * @return Error
+     */
+    virtual Error GatheringInstanceData(
+        const InstanceIdent& instance, InstanceMonitoringData* monitoringData) const = 0;
+};
+
+/**
+ * Monitor sender interface.
+ */
+class MonitorSenderItf {
+public:
+    /**
+     * Send monitoring data
```

Comment:
3rd form in comments: `Sends` etc. In all cases.

Reply:
Done